packages/deployer/ens.js

```diff
@@ -1,23 +1,22 @@
-const ENSJS = require("ethereum-ens");
+const { getEnsAddress, default: ENSJS } = require("@ensdomains/ensjs");
 const contract = require("@truffle/contract");
 const { sha3 } = require("web3-utils");
 const { hash } = require("eth-ens-namehash");
 
 class ENS {
-  constructor({ provider, ensSettings }) {
-    this.ensSettings = ensSettings;
+  constructor({ provider, networkId, ens }) {
+    this.networkId = networkId;
     this.provider = provider;
     this.devRegistry = null;
+    // we need a reference to the ens field to update it for @truffle/contract
+    this.ens = ens;
   }
 
   determineENSRegistryAddress() {
-    if (this.ensSettings.registryAddress) {
-      return this.ensSettings.registryAddress;
-    } else if (
-      this.ensjs &&
-      this.ensjs.registryPromise._rejectionHandler0._address
-    ) {
-      return this.ensjs.registryPromise._rejectionHandler0._address;
+    if (this.ens.registryAddress) {
```

Comment:
I notice this function doesn't check for a `networkId` and use that.  I'm assuming that's deliberate, and that doing things by network ID is done elsewhere?   Or what?

Edit: Oh, I see, it's only used for determining the address of a registry address to deploy to (if I'm understanding correctly?).  Dunno if it's worth renaming this to make that clearer.

Reply:
Which name specifically are you saying is a candidate for a rename?

Reply:
`determineENSRegistryAddress`

Reply:
Oh, @eggplantzzz has explained, I misunderstood: it doesn't need to check for the network ID because that was already done when setting up `ensjs`.  So in that case you'll get it from the `ensjs`.